config/yaml.go

```diff
@@ -32,31 +32,67 @@ import (
 )
 
 type yamlConfigProvider struct {
-	roots  []*yamlNode
+	root   *yamlNode
 	vCache map[string]Value
 }
 
 var _ Provider = &yamlConfigProvider{}
 
 func newYAMLProviderCore(files ...io.ReadCloser) Provider {
-	roots := make([]*yamlNode, len(files))
+	root := &yamlNode{
+		nodeType: objectNode,
+		key:      Root,
+		value:    make(map[interface{}]interface{}),
+		children: nil,
+	}
 
-	for n, v := range files {
-		if root, err := newyamlNode(v); err != nil {
+	for _, v := range files {
+		tmp := make(map[interface{}]interface{})
+		if err := unmarshalYamlValue(v, tmp); err != nil {
 			panic(err)
-		} else {
-			roots[n] = root
 		}
+
+		root.value = mergeMaps(root.value, tmp)
 	}
 
 	return &yamlConfigProvider{
-		roots:  roots,
+		root:   root,
 		vCache: make(map[string]Value),
 	}
 }
 
-// NewYAMLProviderFromFiles creates a configration provider from a set of YAML file
-// names
+func mergeMaps(dst interface{}, src interface{}) interface{} {
+	if dst == nil {
+		panic("Destination node is nil")
+	}
+
+	if src == nil {
+		return src
+	}
+
+	switch s := src.(type) {
+	case map[interface{}]interface{}:
+		d, ok := dst.(map[interface{}]interface{})
```

Comment:
I'm all for brevity but I think `d` may be a little too short given how far down it's used here.